src/lib.rs

```diff
@@ -403,12 +392,38 @@ mod tests {
         }
     }
 
-    fn test_client<Req: Serialize, Resp: DeserializeOwned>(
+    #[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+    struct Response2(u64);
+
+    struct TestService2();
+
+    impl TestService2 {
+        fn new() -> TestService2 {
+            TestService2()
+        }
+    }
+
+    impl Service for TestService2 {
+        type Req = String;
+        type Resp = String;
+        type Error = String;
+        type Ctx = ();
+
+        fn serve(
+            &self,
+            _ctx: (),
+            req: Self::Req,
+        ) -> BoxStream<'static, Result<Self::Resp, String>> {
```

Comment:
looks like we need to box `Self::Resp` here but that fails with
> `the trait `formats::_::_serde::Serialize` cannot be made into an object
`formats::_::_serde::Serialize` cannot be made into an objectrustc(E0038)`

which https://crates.io/crates/erased-serde might solve.